unison-src/builtin-tests/tests.u

```diff
@@ -1,7 +1,27 @@
-
+use builtin.Universal ==
 
 tests : '{IO,Exception} ()
 tests = Tests.main do
   check "Nat.+" do 1 + 1 == 2 
   check "Nat.*" do 10 * 100 == 1000 
   check "Nat./" do 100 / 10 == 10 
+
+  -- cryptographic hashing
+  check "Sha1 hashBytes" do hashBytes Sha1 (toUtf8 "") == 0xsda39a3ee5e6b4b0d3255bfef95601890afd80709
+  check "Sha2_256 hashBytes" do hashBytes Sha2_256 (toUtf8 "") == 0xse3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
+  check "Sha2_512 hashBytes" do hashBytes Sha2_512 (toUtf8 "") == 0xscf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e
+  check "Sha3_256 hashBytes" do hashBytes Sha3_256 (toUtf8 "") == 0xsa7ffc6f8bf1ed76651c14756a061d662f580ff4de43b49fa82d80a4b80f8434a
+  check "Sha3_512 hashBytes" do hashBytes Sha3_512 (toUtf8 "") == 0xsa69f73cca23a9ac5c8b567dc185a756e97c982164fe25859e0d1dcc1475c80a615b2123af1f5f94c11e3e9402c3ac558f500199d95b6d3e301758586281dcd26
+  check "Blake2s_256 hashBytes" do hashBytes Blake2s_256 (toUtf8 "") == 0xs69217a3079908094e11121d042354a7c1f55b6482ca1a51e1b250dfd1ed0eef9
+  check "Blake2b_256 hashBytes" do hashBytes Blake2b_256 (toUtf8 "") == 0xs0e5751c026e543b2e8ab2eb06099daa1d1e5df47778f7787faab45cdf12fe3a8
+  check "Blake2b_512 hashBytes" do hashBytes Blake2b_512 (toUtf8 "") == 0xs786a02f742015903c6c6fd852552d272912f4740e15847618a86e217f71f5419d25e1031afee585313896444934eb04b903a685b1448b755d56f701afe9be2ce
+
+  -- hmacs
+  check "Sha1 hmacBytes" do hmacBytes Sha1 (toUtf8 "key") (toUtf8 "") == 0xsf42bb0eeb018ebbd4597ae7213711ec60760843f
+  check "Sha2_256 hmacBytes" do hmacBytes Sha2_256 (toUtf8 "key") (toUtf8 "") == 0xs5d5d139563c95b5967b9bd9a8c9b233a9dedb45072794cd232dc1b74832607d0
+  check "Sha2_512 hmacBytes" do hmacBytes Sha2_512 (toUtf8 "key") (toUtf8 "") == 0xs84fa5aa0279bbc473267d05a53ea03310a987cecc4c1535ff29b6d76b8f1444a728df3aadb89d4a9a6709e1998f373566e8f824a8ca93b1821f0b69bc2a2f65e
+  check "Sha3_256 hmacBytes" do hmacBytes Sha3_256 (toUtf8 "key") (toUtf8 "") == 0xs74f3c030ecc36a1835d04a333ebb7fce2688c0c78fb0bcf9592213331c884c75
+  check "Sha3_512 hmacBytes" do hmacBytes Sha3_512 (toUtf8 "key") (toUtf8 "") == 0xs7539119b6367aa902bdc6f558d20c906d6acbd4aba3fd344eb08b0200144a1fa453ff6e7919962358be53f6db2a320d1852c52a3dea3e907070775f7a91f1282
+  check "Blake2s_256 hmacBytes" do hmacBytes Blake2s_256 (toUtf8 "key") (toUtf8 "") == 0xs67148074efc0f6741b474ef81c4d98d266e880d372fe723d2569b1d414d234be
+  check "Blake2b_256 hmacBytes" do hmacBytes Blake2b_256 (toUtf8 "key") (toUtf8 "") == 0xs4224e1297e51239a642e21f756bde2785716f872298178180d7f3d1d36a5e4e4
+  check "Blake2b_512 hmacBytes" do hmacBytes Blake2b_512 (toUtf8 "key") (toUtf8 "") == 0xs019fe04bf010b8d72772e6b46897ecf74b4878c394ff2c4d5cfa0b7cc9bbefcb28c36de23cef03089db9c3d900468c89804f135e9fdef7ec9b3c7abe50ed33d3
```

Comment:
I'd convert these to use `===` and merge with trunk, which has fixes to abilities implementation, so the test suite should actually be able to run.